### ISSUE TITLE
False-positive in * .js files

### ISSUE THREAD

**Issue by xereda** [stylelint/stylelint#3434]
False-positive in * .js files.

![stylelint-unknown-word](https://user-images.githubusercontent.com/14337850/42093440-acae62c8-7b82-11e8-8c3e-72ea65abad88.png)

> Which rule, if any, is the bug related to?

Unknown word (CssSyntaxError)

> What CSS is needed to reproduce the bug?

file.js
```js
import Vue from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import './registerServiceWorker';

Vue.config.productionTip = false;

new Vue({
  router,
  store,
  render: h => h(App),
}).$mount('#app');

```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "processors": ["stylelint-processor-html"],
  "plugins": [
    "stylelint-order",
    "stylelint-scss",
    "stylelint-declaration-use-variable",
  ],
  "rules": {
    "sh-waqar/declaration-use-variable": "/color/",
    "scss/selector-no-redundant-nesting-selector": true,
    "order/order": [
			"custom-properties",
			"declarations"
		],
		"order/properties-alphabetical-order": true,
    "at-rule-empty-line-before": [
        "always",
        {
            "except": [
                "blockless-after-same-name-blockless",
                "first-nested"
            ],
            "ignore": [
                "after-comment"
            ]
        }
    ],
    "at-rule-name-case": "lower",
    "at-rule-name-space-after": "always-single-line",
    "at-rule-semicolon-newline-after": "always",
    "block-closing-brace-empty-line-before": "never",
    "block-closing-brace-newline-after": "always",
    "block-closing-brace-newline-before": "always-multi-line",
    "block-closing-brace-space-before": "always-single-line",
    "block-opening-brace-newline-after": "always-multi-line",
    "block-opening-brace-space-after": "always-single-line",
    "block-opening-brace-space-before": "always",
    "color-named": "never",
    "color-hex-case": "lower",
    "color-hex-length": "short",
    "comment-empty-line-before": [
        "always",
        {
            "except": [
                "first-nested"
            ],
            "ignore": [
                "stylelint-commands"
            ]
        }
    ],
    "comment-whitespace-inside": "always",
    "custom-property-empty-line-before": [
        "always",
        {
            "except": [
                "after-custom-property",
                "first-nested"
            ],
            "ignore": [
                "after-comment",
                "inside-single-line-block"
            ]
        }
    ],
    "declaration-bang-space-after": "never",
    "declaration-bang-space-before": "always",
    "declaration-block-semicolon-newline-after": "always-multi-line",
    "declaration-block-semicolon-space-after": "always-single-line",
    "declaration-block-semicolon-space-before": "never",
    "declaration-block-single-line-max-declarations": 1,
    "declaration-block-trailing-semicolon": "always",
    "declaration-colon-newline-after": "always-multi-line",
    "declaration-colon-space-after": "always-single-line",
    "declaration-colon-space-before": "never",
    "declaration-empty-line-before": [
        "always",
        {
            "except": [
                "after-declaration",
                "first-nested"
            ],
            "ignore": [
                "after-comment",
                "inside-single-line-block"
            ]
        }
    ],
    "function-comma-newline-after": "always-multi-line",
    "function-comma-space-after": "always-single-line",
    "function-comma-space-before": "never",
    "function-max-empty-lines": 0,
    "function-name-case": "lower",
    "function-parentheses-newline-inside": "always-multi-line",
    "function-parentheses-space-inside": "never-single-line",
    "function-whitespace-after": "always",
    "indentation": 2,
    "length-zero-no-unit": true,
    "max-empty-lines": 1,
    "media-feature-colon-space-after": "always",
    "media-feature-colon-space-before": "never",
    "media-feature-name-case": "lower",
    "media-feature-parentheses-space-inside": "never",
    "media-feature-range-operator-space-after": "always",
    "media-feature-range-operator-space-before": "always",
    "media-query-list-comma-newline-after": "always-multi-line",
    "media-query-list-comma-space-after": "always-single-line",
    "media-query-list-comma-space-before": "never",
    "no-eol-whitespace": true,
    "no-missing-end-of-source-newline": true,
    "number-leading-zero": "never",
    "number-no-trailing-zeros": true,
    "property-case": "lower",
    "property-no-unknown": true,
    "rule-empty-line-before": [
        "always-multi-line",
        {
            "except": [
                "first-nested"
            ],
            "ignore": [
                "after-comment"
            ]
        }
    ],
    "selector-attribute-brackets-space-inside": "never",
    "selector-attribute-operator-space-after": "never",
    "selector-attribute-operator-space-before": "never",
    "selector-combinator-space-after": "always",
    "selector-combinator-space-before": "always",
    "selector-descendant-combinator-no-non-space": true,
    "selector-list-comma-newline-after": "always",
    "selector-list-comma-space-before": "never",
    "selector-max-empty-lines": 0,
    "selector-pseudo-class-case": "lower",
    "selector-pseudo-class-parentheses-space-inside": "never",
    "selector-pseudo-element-case": "lower",
    "selector-pseudo-element-colon-notation": "double",
    "selector-type-case": "lower",
    "unit-case": "lower",
    "value-list-comma-newline-after": "always-multi-line",
    "value-list-comma-space-after": "always-single-line",
    "value-list-comma-space-before": "never",
    "value-list-max-empty-lines": 0
  }
}
```

> Which version of stylelint are you using?

9.3.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

vscode

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, in a js file.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

![error_stylelint](https://user-images.githubusercontent.com/14337850/42093425-a3550452-7b82-11e8-849e-a11be03989db.png)

**Comment by evilebottnawi** [comment]
@xereda we don't support `js` files

**Comment by jeddy3** [comment]
@xereda Thanks for the report.

> we don't support js files

I think there's more going on here.

From what I gather, _unofficial_ support for JS (and a lot of other things) was side-loaded via our `postcss-html` and `postcss-syntax` dependencies.

The author of the Visual Studio Code stylelint extension proactively updated the extension to run stylelint for [all these new file types](https://github.com/shinnn/vscode-stylelint#usage). 

I was (pleasantly) surprised when I noticed the style objects, i.e. `style={{}}`, in my React code being linted in Visual Studio Code.

I'm grateful that the author of the extension was proactive as seeing this linting in action has convinced me that _offical_ out-of-the-box support for CSS-in-JS is the way to go forward.

We can pick this up in https://github.com/stylelint/stylelint/issues/3264.

Judging by @xereda report, there are a few issues to work out first, though.

@xereda I believe the extension will respect the [`ignoreFiles`](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#ignorefiles) configuration object property and the [`.stylelintignore` file](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#stylelintignore). So, you can put `**/*.js` in either to turn off linting for JavaScript files for now. 

I'll leave this issue open so that we can track the underlying problem, which is likely in postcss-syntax. @gucong3000 Any ideas what it might be?

**Comment by xereda** [comment]
@jeddy3 I added the `.stylelintignore` file with the following content:

```
**/*.js
```

But it did not work.

**Comment by hudochenkov** [comment]
Yeah, VSCode plugin doesn't support `.stylelintignore` (and it isn't mentioned in its readme). Try to exclude .js files with [`ignoreFiles`](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#ignorefiles) in your stylelint config.

```
{
  rules: {},
  ignoreFiles: [
    '**/*.js'
  ]
}
```

**Comment by xereda** [comment]
tks @hudochenkov

**Comment by shinnn** [comment]
```diff
- VSCode plugin doesn't support .stylelintignore
+ stylelint API doesn't support `.stylelintignore`
```

https://github.com/stylelint/stylelint/issues/2833

**Comment by shinnn** [comment]
> it isn't mentioned in its readme

It should be mentioned in *stylelint's* readme. Since vscode-stylelint document already have a link to https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md, a stylelint maintainer adds a mention to this bug neglected for a long time, users will soon notice it.

How will vscode-stylelint document be bloated when it contains a list of all the stylelint bugs?

**Comment by gucong3000** [comment]
@xereda
Remove `processors` from the configuration can fix this error.

`stylelint-processor-html` is not compatible with JS, and `stylelint` currently has built-in support for HTML.

**Comment by TimBa1** [comment]
@xereda please I am having the same issue. how did you solve yours?
